### PR TITLE
implement yaml

### PR DESCRIPTION
**Pull request by kaplanelad**
Implementing yaml

functions:
[yaml.is_valid](https://www.openpolicyagent.org/docs/latest/policy-reference/#builtin-encoding-yamlis_valid)
[yaml.marshal](https://www.openpolicyagent.org/docs/latest/policy-reference/#builtin-encoding-yamlmarshal)
[yaml.is_valid](https://www.openpolicyagent.org/docs/latest/policy-reference/#builtin-encoding-yamlunmarshal)

### CODE REVIEW

**Comment by kaplanelad** [comment]
Hey @sandhose, 
Run Clippy ci fails on too many lines in `src/builtins/mod.rs` file. 
should I use `#[deny(clippy::too_many_lines)]` on top of `resolve` function?

**Comment by sandhose** [comment]
> Run Clippy ci fails on too many lines in `src/builtins/mod.rs` file. 
> 
> should I use `#[deny(clippy::too_many_lines)]` on top of `resolve` function?

Sure, go ahead! I did the same in #18

Note that I'm currently out of office until August 1st, so I'll probably review this PR some time next week

**Comment by kaplanelad** [comment]
> Note that I'm currently out of office until August 1st, so I'll probably review this PR some time next week

Sure no worries, have fun :)

File: tests/infra-fixtures/test-yaml.rego [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package test
+
+is_valid := yaml.is_valid("--\nfoo: bar num: 1\n")
+
+is_valid_1 := yaml.is_valid("foo: bar\nnum: 1")
+
+marshal := yaml.marshal("foo: bar\nnum: 1")
```
Comment (sandhose) [inline tests/infra-fixtures/test-yaml.rego:7]: I think a better test would be
```suggestion
marshal := yaml.marshal({"foo": "bar", "num": 1})
```

**Review by sandhose — CHANGES_REQUESTED** [review]
Looks good overall

Could you please signoff your PR as explained in [`CONTRIBUTING.md`](https://github.com/matrix-org/rust-opa-wasm/blob/main/CONTRIBUTING.md#sign-off)?

**Comment by kaplanelad** [comment]
@sandhose following rebase merge --signoff problems and conflicts, the PR included changes that not related to the current yaml pr. 
i have open a new one 
https://github.com/matrix-org/rust-opa-wasm/pull/44

**Pull request closed without merging**